### DISCUSSION TITLE
How to draw scatter+line facet chart in altair

### DISCUSSION THREAD

**Discussion by HotWesley** [vega/altair#3499] [Q&A] [answered]
How to draw scatter+line facet chart in altair

below is my code
`def correlation_result_plot(source):
    
    chart_title = alt.TitleParams('-'.join([source['item_id1'].iloc[0], source['item_id2'].iloc[0]]), anchor='middle')
    source['key'] = source['item_id1'].astype(str) + '-' + source['item_id2'].astype(str)
    chart = alt.Chart(source).mark_point().encode(
            x=alt.X('value1', type='quantitative', scale=alt.Scale(zero=False)),
            y=alt.Y('value2', type='quantitative', scale=alt.Scale(zero=False)),
            color=alt.Color('item_id2', legend=None)).properties(width=230, height=230).resolve_scale(x='independent', y='independent')
    
    line = chart.transform_regression('value1', 'value2').mark_line(color='red')
    
    chart = chart.facet(column='key', columns=4)

    return (chart + line).properties(title=chart_title)`

then an error has occurred below
`raise ValueError(
ValueError: Faceted charts cannot be layered. Instead, layer the charts before faceting.`

how can i resolve it.... thanks in advance!

**Comment by joelostblom** [comment, marked as answer]
The error message points you in the right direction: you need to do faceting after layering, rather than before. For your spec, that means adding `line` and `chart` together before using the `facet` method on the layered spec. See this section of the docs for an example https://altair-viz.github.io/user_guide/compound_charts.html#faceted-charts

Duplicate of https://github.com/vega/altair/issues/3310

**Reply by HotWesley** [reply]
thank you for your answer. I modify my code like beloq.
`def test(source):
    
    chart_title = alt.TitleParams('-'.join([source['item_id1'].iloc[0], source['item_id2'].iloc[0]]), anchor='middle')
    source['key'] = source['item_id1'].astype(str) + '-' + source['item_id2'].astype(str)
    chart = alt.Chart(source).mark_point().encode(
            x=alt.X('value1', type='quantitative', scale=alt.Scale(zero=False)),
            y=alt.Y('value2', type='quantitative', scale=alt.Scale(zero=False)),
            color=alt.Color('item_id2', legend=None)).properties(width=230, height=230).resolve_scale(x='independent', y='independent')
    
    line = chart.transform_regression('value1', 'value2').mark_line(color='red')
    
    return alt.layer(chart, line).facet('item_id2:N', columns=4)`

then this error has occurred.
`altair.utils.schemapi.SchemaValidationError: Multiple errors were found.

Error 1: `Spec` has no parameter named 'mark'

    Existing parameter names are:
    args

    See the help for `Spec` to read the full description of these parameters

Error 2: `Spec` has no parameter named 'mark'

    Existing parameter names are:
    args

    See the help for `Spec` to read the full description of these parameters`

**Reply by joelostblom** [reply]
Interesting, I think you found a bug which I reported in more detail https://github.com/vega/altair/issues/3502. As a workaround, you can move `resolve_scale` outside `layer` or `facet` (depending on what you want to resolve, likey `facet`)
